src/main/java/rocks/cleanstone/player/listener/PlayerMoveChunkLoadListener.java

```diff
@@ -1,51 +1,51 @@
 package rocks.cleanstone.player.listener;
 
-import org.apache.commons.lang3.tuple.Pair;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.context.event.EventListener;
 import org.springframework.scheduling.annotation.Async;
 import org.springframework.stereotype.Component;
 import rocks.cleanstone.game.Position;
-import rocks.cleanstone.game.world.chunk.ChunkService;
+import rocks.cleanstone.game.world.chunk.ChunkCoords;
+import rocks.cleanstone.game.world.chunk.NearbyChunkRetriever;
 import rocks.cleanstone.player.Player;
 import rocks.cleanstone.player.PlayerChunkLoadService;
 import rocks.cleanstone.player.event.PlayerMoveEvent;
 import rocks.cleanstone.player.event.PlayerQuitEvent;
 
-import java.util.List;
+import java.util.Collection;
 import java.util.Map;
 import java.util.UUID;
 import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.atomic.AtomicInteger;
-import java.util.stream.Collectors;
 
 @Component
 public class PlayerMoveChunkLoadListener {
 
     private final Map<UUID, AtomicInteger> updateCounterMap = new ConcurrentHashMap<>();
     private final PlayerChunkLoadService playerChunkLoadService;
-    private final ChunkService chunkService;
+    private final NearbyChunkRetriever nearbyChunkRetriever;
     private final Logger logger = LoggerFactory.getLogger(getClass());
 
     @Autowired
-    public PlayerMoveChunkLoadListener(PlayerChunkLoadService playerChunkLoadService, ChunkService chunkService) {
+    public PlayerMoveChunkLoadListener(PlayerChunkLoadService playerChunkLoadService, NearbyChunkRetriever nearbyChunkRetriever) {
         this.playerChunkLoadService = playerChunkLoadService;
-        this.chunkService = chunkService;
+        this.nearbyChunkRetriever = nearbyChunkRetriever;
     }
 
     @Async("playerExec")
     @EventListener
     public void onPlayerMove(PlayerMoveEvent playerMoveEvent) {
         final int chunkX = playerMoveEvent.getNewPosition().getXAsInt() >> 4;
         final int chunkZ = playerMoveEvent.getNewPosition().getZAsInt() >> 4;
+        ChunkCoords coords = ChunkCoords.of(chunkX, chunkZ);
```

Comment:
Was wäre mit einer ChunkCoords.of(Position p) factory, dann wären auch nicht so viele `>> 4` in der Anwendung verstreut, sondern an einer zentralen Stelle.